.cascade-code/Chapman.edu/_components/left_nav/left_nav.vtl

```diff
@@ -102,12 +103,13 @@
         ## By default, set visibility to false
         #set ( $isVisibleChild = false )
         #if ( $asset.assetType == 'folder' && $asset.name != '_files' )
-            ## Both the folder and index page must be visible
             #set ( $folderVisible = $asset.metadata.getDynamicField('Hide from navigation').value != "Yes" )
             #set ( $folderIndexPage = $_.locatePage("${asset.path}/index" ) )
             #set ( $folderIndexVisible = $folderIndexPage.metadata.getDynamicField('Hide from navigation').value != "Yes" )
             
-            #if ( $folderIndexVisible && $folderVisible )
+            ## Both the folder and index page must be visible
+            ## Also make sure this asset's index page was found correctly
+            #if ( $folderIndexVisible && $folderVisible && ( $folderIndexPage.path == "${asset.path}/index" ) )
```

Comment:
How about this approach to explain what `$folderIndexPage.path == "${asset.path}/index"` means.

```
              #set ( $folderVisible = $asset.metadata.getDynamicField('Hide from navigation').value != "Yes" )
              #set ( $folderIndexPage = $_.locatePage("${asset.path}/index") )
              #set ( $folderIndexPageExists = $folderIndexPage.path == "${asset.path}/index" )
              #set ( $folderIndexVisible = $folderIndexPageExists && $folderIndexPage.metadata.getDynamicField('Hide from navigation').value != "Yes" )

              #if ( $folderVisible && $folderIndexVisible )
```

You'd have to change it a couple other places, too.